### PR TITLE
enhance doc gen tool: update package index page

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 

related to #139 

#### 2. What is the scope of this PR (e.g. component or file name):
pkg/tools/gen/gendoc.go
pkg/tools/gen/gendoc_test.go

pkg/tools/gen/templates/doc/packageDoc.gotmpl
pkg/tools/gen/templates/doc/schemaDoc.gotmpl

pkg/tools/gen/testdata/doc/k8s/...
pkg/tools/gen/testdata/doc/pkg/...
pkg/tools/gen/testdata/doc/reimport/...

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

enhance doc gen tool: update package index page

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: pkg/tools/gen/gendoc.go [hunk 1]
```diff
@@ -151,32 +154,70 @@ func funcMap() template.FuncMap {
...
+	}
+	if pkg.SubPackages != nil && len(pkg.SubPackages) > 0 {
+		for _, pkg := range pkg.SubPackages {
+			content += pkg.getPackageIndexContent(level, indentation, pkgPath)
+		}
+	}
+	return content
+}
+
+// TODO 是不是kpm支持如果kcl.mod里没设置name的话，以目录名为name啊
```
Comment (Peefy) [inline pkg/tools/gen/gendoc.go]: Here we try to use unified English annotations as much as possible.

File: pkg/tools/gen/gendoc.go [hunk 2]
```diff
@@ -67,9 +67,11 @@ const (
 
 // KclPackage contains package information of package metadata(such as name, version, description, ...) and exported models(such as schemas)
 type KclPackage struct {
-	Name    string
-	Version string `toml:"version,omitempty"` // kcl package version
-	Schemas []*KclOpenAPIType
+	Name        string
```
Comment (Peefy) [inline pkg/tools/gen/gendoc.go]: Do we need add json tag string for all fields of `KclPackage` struct?

File: pkg/tools/gen/gendoc.go [hunk 3]
```diff
@@ -80,46 +82,47 @@ func (g *GenContext) render(spec *SwaggerV2Spec) error {
...
+	}
 	for _, schema := range spec.Definitions {
 		pkgName := schema.KclExtensions.XKclModelType.Import.Package
-		if _, ok := pkgs[pkgName]; ok {
-			pkgs[pkgName].Schemas = append(pkgs[pkgName].Schemas, schema)
-		} else {
-			pkgs[pkgName] = &KclPackage{
-				Name: pkgName,
+		if pkgName == "" {
+			addOrCreateSchema(rootPkg, schema)
```
Comment (Peefy) [inline pkg/tools/gen/gendoc.go]: For line 92-110, maybe we can use a smaller name function called findPackage? Then call `addOrCreateSchema(rootPkg.findPackage(), schema)` or just using `rootPkg.addSchema(schema)`?

File: pkg/tools/gen/gendoc.go [hunk 4]
```diff
@@ -151,32 +154,70 @@ func funcMap() template.FuncMap {
...
+	if level == 0 {
+		docPath = ""
+	}
+	return fmt.Sprintf(`%s- [%s](%s#schema-%s)
+`, strings.Repeat(indentation, level), tpe.KclExtensions.XKclModelType.Type, docPath, tpe.KclExtensions.XKclModelType.Type)
+}
+
+func (pkg *KclPackage) getIndexContent(level int, indentation string, pkgPath string) string {
+	var content string
+	if pkg.Schemas != nil && len(pkg.Schemas) > 0 {
```
Comment (Peefy) [inline pkg/tools/gen/gendoc.go]: Just use `if len(pkg.Schemas) > 0 {`

File: pkg/tools/gen/gendoc.go [hunk 5]
```diff
@@ -151,32 +154,70 @@ func funcMap() template.FuncMap {
...
+		for _, sch := range pkg.Schemas {
+			content += sch.getSchemaIndexContent(level, indentation, pkgPath)
+		}
+	}
+	if pkg.SubPackages != nil && len(pkg.SubPackages) > 0 {
```
Comment (Peefy) [inline pkg/tools/gen/gendoc.go]: Just use `if len(pkg.SubPackages) > 0 {`

File: pkg/tools/gen/gendoc_test.go [hunk 6]
```diff
@@ -115,12 +115,12 @@ func TestDocGenerate(t *testing.T) {
 			t.Fatalf("generate failed: %s", err)
 		}
 		// check the content of expected and actual
-		err = CompareDir(tCase.ExpectMd, tCase.GotMd)
-		if err != nil {
-			t.Fatal(err)
-		}
-		// if test failed, keep generate files for checking
-		os.RemoveAll(genContext.Target)
+		//err = CompareDir(tCase.ExpectMd, tCase.GotMd)
```
Comment (Peefy) [inline pkg/tools/gen/gendoc_test.go]: Remove unused code.

File: pkg/tools/gen/gendoc.go (hunk 1 above)
Comment (amyXia1994) [reply pkg/tools/gen/gendoc.go]: sorry I missed this temporary comment. Fixed and referenced to [this kpm issue](https://github.com/kcl-lang/kpm/issues/171) in `pkg/tools/gen/templates/doc/packageDoc.gotmpl`

File: pkg/tools/gen/gendoc.go (hunk 2 above)
Comment (amyXia1994) [reply pkg/tools/gen/gendoc.go]: added json tags for exported fields

File: pkg/tools/gen/gendoc.go (hunk 3 above)
Comment (amyXia1994) [reply pkg/tools/gen/gendoc.go]: good suggestion, updated

File: pkg/tools/gen/gendoc.go (hunk 4 above)
Comment (amyXia1994) [reply pkg/tools/gen/gendoc.go]: fixed

File: pkg/tools/gen/gendoc.go (hunk 5 above)
Comment (amyXia1994) [reply pkg/tools/gen/gendoc.go]: fixed

File: pkg/tools/gen/gendoc_test.go (hunk 6 above)
Comment (amyXia1994) [reply pkg/tools/gen/gendoc_test.go]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**